modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ZonePartitionReplicaListener.java

```diff
@@ -131,4 +131,19 @@ public void addTableReplicaListener(TablePartitionId partitionId, Function<RaftC
     public Map<TablePartitionId, ReplicaListener> tableReplicaListeners() {
         return replicas;
     }
+
+    @Override
+    public void onShutdown() {
+        replicas.forEach((id, listener) -> {
```

Comment:
Should we guard it with busy lock?

Reply:
Our shutdowns already covered by busyLock earlier, that's the current pattern I thought

Reply:
Yep, but this shutdown called from ZonePartitionReplicaImpl#shutdown, which called as a usual replica shutdown inside the ReplicaManager

Reply:
Discussed in private. Replica.busyLock should be enough here.